src/vm/comcallablewrapper.cpp

```diff
@@ -2241,11 +2234,11 @@ ComCallWrapper* ComCallWrapper::CopyFromTemplate(ComCallWrapperTemplate* pTempla
     // num interfaces on the object    
     size_t numInterfaces = pTemplate->GetNumInterfaces();
 
-    // we have a template, create a wrapper and initialize from the template
-    // alloc wrapper, aligned 32 bytes
-    if (pWrapperCache->IsDomainUnloading())
+    if (pWrapperCache->IsLoaderAllocatorUnloading())
```

Comment:
> f there are COM references, they keep the LoaderAllocator alive.

So why do we need to throw AppDomainUnloadedException here?

Reply:
This is for other purpose. After the unloading is initiated, this code prevents new CCW creation. But the ones that were created before that still work.

Reply:
Regular P/Invoke interop is allowed even after the unloading is initialized. Why do we want to disallow COM?

Reply:
I don't have a strong opinion on this. This was the way the AppDomain behaved after the Unload was initiated on them, so I have kept it. But I don't think there would be any issue if we removed this check.

Reply:
I still have doubts about this. I do not think we should be blocking any interop when the stuff is unloading.

Reply:
AppDomains had to do it because of they would not be able to forcefully unload otherwise. ALCs do not have forceful unload, so they do not need it.